osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -109,7 +109,15 @@ public string ModsJson
         public string UserString
         {
             get => User?.Username;
-            set => User = new User { Username = value };
+            set => User = new User { Username = value, Id = UserID};
+        }
+
+        [JsonIgnore]
+        [Column("UserID")]
+        public long UserID
```

Comment:
also should this not be nullable? do we actually want to store `1` to the database for users which we don't know IDs?

Reply:
I think in the scenario where the user has never logged in, the best course of action is to store the default initialized userID (which is 1) which is at least recognizeable? To be honest I'm not sure, I've never had experience with a precedent.

Reply:
That's not the only time a `ScoreInfo` is created. What about replay imports, for example?

Reply:
Should be nonnullable